src/vm/sbin/vmadmd.js

```diff
@@ -116,6 +118,66 @@ function zonecfg(args, callback)
     });
 }
 
+/*
+ * This function gets the runtime of the last run of a stopped zone by comparing
+ * the mtime of the /lastbooted and /lastexited files.
+ *
+ * If there are no errors, the last_runtime property will be added to the vmobj.
+ *
+ */
+function addLastRuntime(vmobj, opts, callback)
+{
+    var lastbooted_filename;
+    var lastbooted_mtime;
+    var lastexited_filename;
+    var lastexited_mtime;
+    var zonepath = vmobj.zonepath;
+
+    assert(log, 'missing log'); // log is GLOBAL!
+    assert(zonepath, 'missing zonepath');
+
+    lastbooted_filename = path.join(vmobj.zonepath, '/lastbooted');
+    lastexited_filename = path.join(vmobj.zonepath, '/lastexited');
```

Comment:
The leading `/` for the last path components should not be used, as `path.join` already does that formatting, and it would defeat the purpose of using the cross-platform `path.join`.